lib/src/serializer.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// serializer.dart
+// A serialization block, serializing wide input data onto a narrower channel.
+//
+// 2024 August 27
+// Author: desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// Serializes wide aggregated data onto a narrower serialization stream
+class Serializer extends Module {
+  /// Clk input
+  @protected
+  Logic get clk => input('clk');
```

Comment:
in general (for both components` you're not using these protected input getters.  are you putting them there for future/user extension of these classes?

Reply:
Should I eliminate the getters or the protected attributes?